src/components/Notifications/RuleSimple/RuleSimple.js

```diff
@@ -333,31 +473,114 @@ export default class RuleSimple extends React.Component {
 
     return
   }
+  isValidSecondQueryResponse = () => {
+    return (
+      this.state.secondQueryResponse &&
+      Object.keys(this.state.secondQueryResponse).length > 0 &&
+      this.state.secondQueryResponse.data
+    )
+  }
+  getCompareColumnIndex = (response, storedInitialData, index) => {
+    const selectedNumberColumnName = storedInitialData?.[index]?.compare_column ?? ''
+    let compareColumnIndex
+
+    if (selectedNumberColumnName) {
+      compareColumnIndex = response?.data?.data?.columns?.findIndex((col) => col.name === selectedNumberColumnName)
+    } else {
+      compareColumnIndex = response?.data?.data?.columns?.findIndex((col) => isColumnNumberType(col) && col.is_visible)
+    }
+
+    return compareColumnIndex === -1 ? undefined : compareColumnIndex
+  }
+  getQuery = () => {
+    this.setState({ isLoadingFirstQuery: true, isLoadingSecondQuery: true })
+    const fetchFirstQuery = this.props.queryResponse
+      ? Promise.resolve(this.props.queryResponse)
+      : runQueryOnly({
+          query: this.props.initialData?.[0]?.term_value,
+          ...getAuthentication(this.props.authentication),
+          ...getAutoQLConfig(this.props.autoQLConfig),
+          source: 'data_alert_first_query',
+          pageSize: 2,
+          allowSuggestions: false,
+        })
+    const fetchSecondQuery =
```

Comment:
This complexity is a little tough to read. Is there a way to simplify this?

Reply:
I added a shouldFetchSecondQuery to make it more readable